cpp/tests/sorted-merge/sorted_merge-test.cu

```diff
@@ -0,0 +1,199 @@
+#include <cassert>
+#include <vector>
+#include <memory>
+#include <algorithm>
+#include <gtest/gtest.h>
+#include <nvstrings/NVCategory.h>
+#include <nvstrings/NVStrings.h>
+
+#include "tests/utilities/column_wrapper.cuh"
+#include "tests/utilities/cudf_test_fixtures.h"
+#include "tests/utilities/nvcategory_utils.cuh"
+
+#include <cudf.h>
+#include <cudf/functions.h>
+#include <sorted_merge.hpp>
+#include <rmm/thrust_rmm_allocator.h>
+#include <table.hpp>
+#include "string/nvcategory_util.hpp"
+
+std::vector<gdf_valid_type> bools_to_valids(const std::vector<uint8_t> & input) {
+    std::vector<gdf_valid_type> vec(gdf_valid_allocation_size(input.size()), 0);
+    for(size_t i = 0; i < input.size(); ++i){
+        if(input[i]){
+            gdf::util::turn_bit_on(vec.data(), i);
+        } else {
+            gdf::util::turn_bit_off(vec.data(), i);
+        }
+    }
+    return vec;
+}
+
+template <typename T>
+class SortedMergeTest : public GdfTest {};
+
```

Comment:
Need a test for error cases -- e.g. mismatched column dtypes. Basically anything you have a CUDF_EXPECTS for.